drivers/bluetooth/hci/hci_psoc6_bless.c

```diff
@@ -0,0 +1,260 @@
+		buf = bt_buf_get_evt(hci_rx->data[0], 0, K_NO_WAIT);
+		if (!buf) {
+			LOG_ERR("Failed to allocate the buffer for RX: EVENT ");
+			return;
+		}
+
+		break;
+	case PACKET_TYPE_HCI_ACL_DATA:
+		buf = bt_buf_get_rx(BT_BUF_ACL_IN, K_NO_WAIT);
+		if (!buf) {
+			LOG_ERR("Failed to allocate the buffer for RX: ACL ");
+			return;
+		}
+		bt_buf_set_type(buf, BT_BUF_ACL_IN);
+
+		break;
+
+	default:
+		LOG_WRN("Unsupported HCI Packet Received");
+		return;
+	}
+
+	buf_tailroom = net_buf_tailroom(buf);
+	if (buf_tailroom < hci_rx->dataLength) {
+		LOG_WRN("Not enough space for rx data");
+		return;
+	}
+	net_buf_add_mem(buf, hci_rx->data, hci_rx->dataLength);
+	bt_recv(buf);
+}
+
+static int psoc6_bless_open(void)
+{
+	k_tid_t tid;
+
+	tid = k_thread_create(&psoc6_bless_rx_thread_data, psoc6_bless_rx_thread_stack,
+			      K_KERNEL_STACK_SIZEOF(psoc6_bless_rx_thread_stack),
+			      psoc6_bless_rx_thread, NULL, NULL, NULL,
+			      K_PRIO_COOP(CONFIG_BT_RX_PRIO), 0, K_NO_WAIT);
+	k_thread_name_set(tid, "psoc6_bless_rx_thread");
+
+	return 0;
+}
+
+static int psoc6_bless_send(struct net_buf *buf)
+{
+	cy_en_ble_api_result_t result;
+
+	memset((void *)&hci_tx_pkt, 0, sizeof(cy_stc_ble_hci_tx_packet_info_t));
+
+	hci_tx_pkt.dataLength = buf->len;
+	hci_tx_pkt.data = buf->data;
+
+	switch (bt_buf_get_type(buf)) {
+	case BT_BUF_ACL_OUT:
+		hci_tx_pkt.packetType = PACKET_TYPE_HCI_ACL_DATA;
+		break;
+	case BT_BUF_CMD:
+		hci_tx_pkt.packetType = PACKET_TYPE_HCI_COMMAND;
+		break;
+
+	default:
+		net_buf_unref(buf);
+		return -ENOTSUP;
+	}
+
+	if (k_sem_take(&psoc6_bless_operation_sem, K_MSEC(BLE_LOCK_TMOUT_MS)) != 0) {
+		LOG_ERR("Failed to acquire BLE DRV Semaphore");
+		net_buf_unref(buf);
+		return -EIO;
+	}
+
+	result = Cy_BLE_SoftHciSendAppPkt(&hci_tx_pkt);
+	if (result != CY_BLE_SUCCESS) {
+		LOG_ERR("Error in sending packet reason %d\r\n", result);
+	}
+
+	k_sem_give(&psoc6_bless_operation_sem);
+
+	net_buf_unref(buf);
+
+	return 0;
+}
+
+static int psoc6_bless_setup(void)
+{
+#define BT_HCI_VND_OP_SET_PUBLIC_ADDR BT_OP(0xFD, 0xA0)
+
+	struct net_buf *buf;
+	int err;
+	uint8_t *addr = (uint8_t *)&SFLASH_BLE_DEVICE_ADDRESS[0];
+	uint8_t hci_data[] = {addr[5], addr[4], addr[3],          addr[2],
+			      addr[1], addr[0], BT_ADDR_LE_PUBLIC};
+
+	buf = bt_hci_cmd_create(BT_HCI_VND_OP_SET_PUBLIC_ADDR, sizeof(hci_data));
+	if (buf == NULL) {
+		LOG_ERR("Unable to allocate command buffer");
+		return -ENOMEM;
+	}
+
+	/* Add data part of packet */
+	net_buf_add_mem(buf, hci_data, sizeof(hci_data));
+
+	err = bt_hci_cmd_send_sync(BT_HCI_VND_OP_SET_PUBLIC_ADDR, buf, NULL);
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+static int psoc6_bless_hci_init(void)
+{
+	cy_en_ble_api_result_t result;
+	static const struct bt_hci_driver drv = {.name = "PSoC 6 BLESS",
+						 .bus = BT_HCI_DRIVER_BUS_VIRTUAL,
+						 .quirks = BT_QUIRK_NO_RESET,
+						 .open = psoc6_bless_open,
+						 .send = psoc6_bless_send,
+						 .setup = psoc6_bless_setup};
```

Comment:
You had this properly formatted earlier.. not sure why you changed it. Anyway, please make it like this:
```c
static const struct bt_hci_driver drv = {
	.name = "PSoC 6 BLESS",
	...
};
```

Reply:
it was formatted by clang-format, looks like a trailing comma is required for it to work properly, updated the other structures as well..